lib/features/modeling/behavior/CreateChoreoTaskBehavior.js

```diff
@@ -69,183 +68,98 @@ function chooseParticipants(hints, allParticipants) {
-      });
-      let shapeB = elementFactory.createShape(assign({
-        type: 'bpmn:Participant',
-        id: model.ids.nextPrefixed('ParticipantBand_', partB),
-        businessObject: partB,
-        diBand: diB,
-        activityShape: shape
-      }, boundsB));
-
-      shape.bandShapes = [shapeA, shapeB];
-      // remember some stuff in the context for later linking
-      assign(context, {
-        createdShapeA: shapeA,
-        createdShapeB: shapeB
-      });
-
-      // create initiating message flow and shape (invisible)
-      if (is(shape, 'bpmn:ChoreographyTask')) {
-        let messageFlow = createMessageFlowSemantics(
-          injector,
-          activity,
-          activity.get('initiatingParticipantRef')
-        );
-        let messageShape = createMessageShape(injector, shapeA, messageFlow);
-        messageShape.hidden = true;
-        assign(context, {
-          createdMessageFlow: messageFlow,
-          createdMessageShape: messageShape
-        });
+    const context = event.context;
+    const shape = context.shape;
+    if (is(shape, 'bpmn:ChoreographyActivity') && shape.type !== 'label') {
+      if (!shape.businessObject.name) {
+        shape.businessObject.name = 'New Activity';
       }
     }
   });
 
-
-  this.executed('shape.create', function(event) {
+  this.postExecuted('shape.create', function(event) {
     let context = event.context;
     let shape = context.shape;
+    const businessObject = shape.businessObject;
 
-    // Same as above: new vs. copied
-    if (is(shape, 'bpmn:ChoreographyActivity') && shape.type !== 'label' && !shape.oldBusinessObject) {
-      // add the bands to the BPMNDiagram semantic object
-      let choreo = canvas.getRootElement().businessObject;
-      let bpmnDiagram = choreo.di;
-      bpmnDiagram.get('planeElement').push(context.createdShapeA.diBand);
-      bpmnDiagram.get('planeElement').push(context.createdShapeB.diBand);
-      context.createdShapeA.diBand.$parent = bpmnDiagram;
-      context.createdShapeB.diBand.$parent = bpmnDiagram;
-
-      // add bands to canvas
-      canvas.addShape(context.createdShapeA, shape);
-      canvas.addShape(context.createdShapeB, shape);
-
-      // properly link elements regarding choreography tasks
-      if (is(shape, 'bpmn:ChoreographyTask')) {
-        linkMessageFlowSemantics(injector, shape.businessObject, context.createdMessageFlow);
-        canvas.addShape(context.createdMessageShape, context.createdShapeA);
+    if (is(shape, 'bpmn:ChoreographyActivity') &&
+      shape.type !== 'label'
+    ) {
+
+      let participants = [];
+      if (context.hints.createElementsBehavior === false) {
+        // the shape was copied
+        participants = [].concat(businessObject.participantRef);
+        businessObject.participantRef = []; // will be set by bandhandler
+      } else {
+        const choreo = canvas.getRootElement();
+        const recommended = chooseParticipants(context.hints, choreo);
```

Comment:
Won't this break if `chooseParticipants` returns `{}`?

Reply:
Good catch! For the record it does not break, but that is probably more a lucky coincidence. I will make the code a bit clearer at this part